query_optimizer/ExecutionGenerator.hpp

```diff
@@ -427,7 +445,7 @@ class ExecutionGenerator {
   /**
    * @brief The cost model to use for estimating aggregation hash table size.
    */
-  std::unique_ptr<cost::CostModel> cost_model_for_aggregation_;
+  std::unique_ptr<cost::StarSchemaSimpleCostModel> cost_model_for_aggregation_;
```

Comment:
We need some extra analysis done in `StarSchemaSimpleCostModel`. I will refactor the analysis part out of `StarSchemaSimpleCostModel` into `AnalysisUtil` in a follow-up PR.